src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -453,6 +453,23 @@ private void OnExecutionMessageReceived(object sender, MessageReceivedEventArgs
 
                         this.channel.Send(data);
                         break;
+
+                    case MessageType.AttachDebuggerToProcess:
+                        var testProcessPid = this.dataSerializer.DeserializePayload<TestProcessAttachDebuggerPayload>(message);
+                        if (!(testRunEventsHandler is ITestRunEventsHandler2))
```

Comment:
same situation with the `is` and `as` here. Also operation is not supported by the current run handler is very vague, if you want to do a check like this, then error out with something like "AttachDebuggerToProcess is only supported in ITestRunEventsHandler2, but it was called with ITestRunEventsHandler." If you are able to use the default implementation this whole checking would go away.

Reply:
Done.